portfolio/src/main/java/com/google/sps/storage/DatastoreCommentStorage.java

```diff
@@ -0,0 +1,83 @@
+package com.google.sps.storage;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.sps.storage.CommentStorage;
+import com.google.sps.data.Comment;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class DatastoreCommentStorage implements CommentStorage {
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  private static final String ENTITY_QUERY = "Comment";
+  private static final String ENTITY_TEXT_PARAM = "comment-text";
+  private static final String ENTITY_NAME_PARAM = "username";
+  private static final String ENTITY_TIME_PARAM = "time-posted";
+
+  @Override
+  public void insert(Comment comment) {
+    String name = comment.name();
+
+    if (comment.name().equals("")) {
+      name = "Anonymous";
```

Comment:
I don't like the storage layer doing this default field resolution. Please move it in one of two ways:
- decide that empty is allowed in your DB and have default resolution on read
- decide that empty is not allowed and either:
  - resolve before inserting to storage
  - reject the request as a user error